implementations/rust/refactor_prototype/message_router/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+#![no_std]
+extern crate alloc;
+use alloc::boxed::Box;
```

Comment:
Probably want all the `use alloc` statements in a cfg_if block since this is not needed for std mode.

Reply:
Good point, there's a lot of work to be done when we configure for std/no_std. This is just a prototype.